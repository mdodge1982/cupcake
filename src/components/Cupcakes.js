import React, {Component} from 'react';
import './Cupcakes.css';
import Cupcake from '../components/Cupcake';

class Cupcakes extends Component {
	constructor() {
		super();
		this.state = {
			int: null
		};
	}
	render() {
		return (
			<ul className="Cupcakes">
				{this.props.cupcakes.map(cake => (
					<Cupcake key={cake.id} {...cake} />
				))}
			</ul>
		);
	}
	componentDidMount() {
		let count = this.props.count;
		this.setState({
			int: setInterval(() => {
				if(count>0){
					this.props.addCupcake();
				}else{
					clearInterval(this.state.int);
				}
				count--;
			},1000)
		});
	}
	componentWillUnmount() {
		clearInterval(this.state.int);
	}
}

export default Cupcakes;
