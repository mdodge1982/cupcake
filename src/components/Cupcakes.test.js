import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Cupcakes from './Cupcakes';

test('Initial render', () => {
	const cupcakeArr = [{id:123}];
	const component = renderer.create(
		<Cupcakes cupcakes={cupcakeArr} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

jest.useFakeTimers();

test('Calls onInterval every second for specified # of times', () => {
	const count = 4;
	const mockFn = jest.fn();
	const component = ReactTestUtils.renderIntoDocument(
		<Cupcakes
		cupcakes={[]}
		addCupcake={mockFn}
		count={count} />
	);
	jest.runAllTimers();
	expect(mockFn.mock.calls.length).toBe(count);
});
