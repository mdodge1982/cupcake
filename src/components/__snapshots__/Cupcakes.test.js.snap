// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial render 1`] = `
<ul
  className="Cupcakes"
>
  <li
    className="Cupcake undefined"
    style={
      Object {
        "bottom": "undefinedpx",
        "height": "undefinedpx",
        "left": "undefinedpx",
        "width": "undefinedpx",
      }
    }
  >
    
  </li>
</ul>
`;
